### PR TITLE
Implement delayed slot in RzIL

### PR DESCRIPTION
**Pull request by wargio**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

This code is not tested.

### CODE REVIEW

File: librz/il/il_vm_eval.c [hunk 1]
```diff
@@ -291,20 +291,34 @@ RZ_API void rz_il_vm_clear_events(RzILVM *vm) {
...
  */
-RZ_API bool rz_il_vm_step(RzILVM *vm, RzILOpEffect *op, ut64 fallthrough_addr) {
+RZ_API bool rz_il_vm_step(RzILVM *vm, RzILOpEffect *op, ut64 fallthrough_addr, ut32 delayed_by) {
 	rz_return_val_if_fail(vm && op, false);
 
 	rz_il_vm_clear_events(vm);
 
 	// Set the successor pc **before** evaluating. Any jmp/goto may then overwrite it again.
-	RzBitVector *next_pc = rz_bv_new_from_ut64(vm->pc->len, fallthrough_addr);
+	RzBitVector *next_pc = NULL;
```
Comment (XVilka) [inline librz/il/il_vm_eval.c:302]: Not very happy about this solution, to be honest

Comment (wargio) [reply librz/il/il_vm_eval.c:302]: i'm exploring possibilities.

**Comment by XVilka** [comment]
@wargio @brightprogrammer as we settled on the reordering solution, could be this closed?

**Comment by brightprogrammer** [comment]
Yes

**Pull request closed without merging**